PC/getpathp.c

```diff
@@ -265,7 +265,21 @@ canonicalize(wchar_t *buffer, const wchar_t *path)
         return _PyStatus_NO_MEMORY();
     }
 
-    if (FAILED(PathCchCanonicalizeEx(buffer, MAXPATHLEN + 1, path, 0))) {
+    if (PathIsRelativeW(path)) {
+        wchar_t buff[MAXPATHLEN];
```

Comment:
This allocation should be `MAXPATHLEN + 1`. In this file, `MAXPATHLEN` refers to the string length of the path, without the terminating null.